### ISSUE TITLE
Dio does not remove content-length header from GET request even if you specify content-length = -1 

### ISSUE THREAD

**Issue by hygehyge** [flutterchina/dio#944]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none
I've found [#891](https://github.com/flutterchina/dio/issues/891), but this has closed by bot and does not solve my problem.

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   1.20.3   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

When I create an HTTP GET request without content-length header as below, dio always adds `content-length:0`, where [http](https://pub.dev/packages/http) package doesn't.
```Dart
//dio makes request as
//user-agent: Dart/2.9 (dart:io)
//accept-encoding: gzip
//content-length: 0  <-- THIS IS THE PROBLEM !!
//host: www.example.com
final options = BaseOptions()
    ..contentType = null
    ..headers[Headers.contentLengthHeader] = '-1';
final dio = Dio(options);
final dioResponse = await dio.get('https://www.example.com/sample_api');

//http makes request as
//user-agent: Dart/2.9 (dart:io)
//accept-encoding: gzip
//host: www.example.com
final client = http.Client();
final httpResponse = await client.get(
    'https://www.example.com/sample_api',
    headers: {
        'content-length': '-1',
    },
);
```

And the server that I want to request returns `Status Code:400 Bad Request` if request's header contains `content-length`.
How can I remove this header ?

### Edit
After some additional research, I've found strange behavior.
In DefaultHttpClientAdapter's fetch method,  if I put a breakpoint on `Future future = request.close();` at line 62 and watch `request` then continue, the request doesn't contain `content-length`.

**io_adapter.dart**
```Dart
HttpClientRequest request;
try {
  request = await requestFuture;
  //Set Headers
  options.headers.forEach((k, v) => request.headers.set(k, v));
} on SocketException catch (e) {
  if (e.message.contains('timed out')) _throwConnectingTimeout();
  rethrow;
}

request.followRedirects = options.followRedirects;
request.maxRedirects = options.maxRedirects;

if (options.method != 'GET' && requestStream != null) {
  // Transform the request data
  await request.addStream(requestStream);
}
Future future = request.close(); //<-- line 62, break here and watch request, the generated request doesn't contain content-length.
```

I put `Future.delayed()` or `print(request)` before `Future future = request.close();`, but it does nothing.
`content-length` is removed only when I watch `request`.
What is the cause of this behavior ?

**Comment by kobe-thiagofontes** [comment]
Issue still persists
